.github/workflows/terraform-observe_scheduler.yaml

```diff
@@ -31,6 +31,42 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+  get-test-directories:
+    if: "!contains(fromJSON(inputs.skip).jobs, 'terraform-tests')"
+    name: Get Test Directories For OPAL Validation
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Build matrix
+        id: matrix
+        shell: bash
+        run: |
+          DIRS="$(find . -path "*.tests/*" ! -path "*.terraform*" -type d )"
```

Comment:
I think we'll end up having to move this logic into the scripts/terraform_tests.sh script.

Reply:
interesting -- why so?

is it that you'd like users to be able to run all the tests locally at the same time? if yes, we could update the script to support an "all" argument that does this extra logic, while otherwise accepting individual paths. 

That way we can keep the build matrix (which i'd like to do, since that means all the test run in parallel quickly)

Reply:
A few reasons:
- tests should be reproducible and portable. Having part of the logic in github actions makes it harder to discern what the expected layout is for example. If we ever decide to use jenkins instead, we have to convert this to some other framework. None of that happens if you restrict things to `make test`
- the logic for for the find may become more convoluted, I'm not sure. Right now we're at the limit of reasonable.
- we can parallelize in the bash script or makefile. It's not as pretty, but totally doable.
- our apiserver may have issues with parallelization. It shouldn't, but it might, so we may have to apply serially anyway.

Reply:
oh interesting, thanks for expanding on that! I'll look into parallelizing in bash/makefile, but i'll keep in mind possible limits of the apiserver on parallelization (maybe i can allow for configurable serial/parallel executions or something)

Reply:
oh cool! then i'll just move forward with this for now :) thanks for clarifying!